server/replicate/replicate.go

```diff
@@ -133,8 +150,90 @@ func (m *ModeManager) drSwitchToSync() error {
 	defer m.Unlock()
 	dr := drAutosyncStatus{State: drStateSync}
 	if err := m.storage.SaveReplicateStatus(modeDRAutosync, dr); err != nil {
+		log.Warn("failed to switch to sync state", zap.String("replicate-mode", modeDRAutosync), zap.Error(err))
 		return err
 	}
 	m.drAutosync = dr
+	log.Info("switched to sync state", zap.String("replicate-mode", modeDRAutosync))
 	return nil
 }
+
+func (m *ModeManager) drGetState() string {
+	m.RLock()
+	defer m.RUnlock()
+	return m.drAutosync.State
+}
+
+const (
+	idleTimeout  = time.Minute
+	tickInterval = time.Second * 10
+)
+
+// Run starts the background job.
+func (m *ModeManager) Run(quit chan struct{}) {
+	select {
+	case <-time.After(idleTimeout):
```

Comment:
Why need to wait for an idle timeout?  better to add a comment.

Reply:
ok, good idea